Marlin/stepper.cpp

```diff
@@ -601,14 +639,28 @@ void Stepper::init() {
   TCNT1 = 0;
   ENABLE_STEPPER_DRIVER_INTERRUPT();
 
-  #if ENABLED(ADVANCE)
+  #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
+
+    #if ENABLED(LIN_ADVANCE)
+
+      for (int i = 0; i < EXTRUDERS; i++) {
```

Comment:
@Jbrazio in order to fix the advance feature, it needs a lot of discussion.
You could create a wiki collaboration page on this.
I have spend the last week a lot of thinking of this feature and looked the
implementations across github like Sailfish (JNK) and Smoothieboard.

The lessons learnt by me are worth discussing on a wiki page. For example,
to give you a small taste: the XY motion is calculated by the planner which
is the old Grbl planner written a long time ago. It was written for CNC
machines. The extruder was added later and needs to be incorporated into
the planner in a smarter and more logical way. The priming and de-priming
of the hot-end need to be ahead of the xy planned motion in terms of
time/steps. Lot's of ideas around that in terms of implementation like a
third pass of the planning cycle to fill the buffer with the extruder
values and using boost/Lapack libraries to simplify calculations etc.

Basically I suggest to take a structured approach in collaboration on a
wiki page for this JNK feature rather than discussing via a issue thread
(you loose the thinking and conclusions and new comers has to scroll
through large threads whilst differentiating between opinions/rubbish and
real facts) and throwing in the implementations from members which I don't
understand. What do you think?

On 5 May 2016 at 09:28, João Brázio <notifications@github.com> wrote:

> In Marlin/stepper.cpp
> <https://github.com/MarlinFirmware/Marlin/pull/3676#discussion_r62131409>:
>
> > @@ -601,14 +639,28 @@ void Stepper::init() {
> >    TCNT1 = 0;
> >    ENABLE_STEPPER_DRIVER_INTERRUPT();
> >
> > -  #if ENABLED(ADVANCE)
> > +  #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
> > +
> > +    #if ENABLED(LIN_ADVANCE)
> > +
> > +      for (int i = 0; i < EXTRUDERS; i++) {
>
> @thinkyhead <https://github.com/thinkyhead> shouldn't we clean up the
> "old" ADVANCE feature ? Having two advance features using two different
> methods but when only one has active support and is known to work.. I would
> cull the other one.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/MarlinFirmware/Marlin/pull/3676/files/b497db094e421700112ec5a07e32a20bb9deaf6c#r62131409>
>